### PR TITLE
Add design documentation for test targets

### PR DESCRIPTION
**Pull request by naamunds**
/cc @weshaggard @chcosta @MattGal

### CODE REVIEW

File: Documentation/test-targets-design.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# Test Target Design
+
+This document describes the design of the BuildTools test targets. For documentation on their usage, see [Test Targets](test-targets-usage.md).
+
+The primary entry points are the [`Test` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L336) in Microsoft.DotNet.Build.Tasks and the [`CloudBuild` target](https://github.com/dotnet/buildtools/blob/73375848b73a586a8ac439a1263b605901414ad8/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets#L79) in Microsoft.DotNet.Build.CloudTestTasks, which are each described below.
```
Comment (MattGal) [inline Documentation/test-targets-design.md]: is referencing a specific version of the file intentional here?  (i.e. to keep the line # accurate or something)

File: Documentation/test-targets-design.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+## `Test` target
+
+The `Test` target is used to run or archive tests locally. At a high level, this target performs four steps:
+
+1. Copies the required files for running the tests to the test execution directory.
+2. Generates the test execution script, which was introduced for running the tests in Helix but is also used for running them locally.
```
Comment (MattGal) [inline Documentation/test-targets-design.md]: Maybe say something more generic than Helix here, which may confuse some users.  Like "For running the tests the same way in distributed automation systems" or something.

File: Documentation/test-targets-design.md (hunk 1 above)
Comment (naamunds) [inline Documentation/test-targets-design.md]: Yes; I figured that was the only feasible way to reference a line number.

File: Documentation/test-targets-design.md (hunk 2 above)
Comment (naamunds) [inline Documentation/test-targets-design.md]: Good idea. I've changed it to "distributed automation systems such as [Helix](https://helix.dot.net/)".

**Pull request merged**